lms/djangoapps/gating/api.py

```diff
@@ -61,13 +61,13 @@ def _get_subsection_percentage(subsection_grade):
     """
     Returns the percentage value of the given subsection_grade.
     """
-    return _calculate_ratio(subsection_grade.graded_total.earned, subsection_grade.graded_total.possible) * 100.0
+    return subsection_grade.percent_graded * 100.0
 
 
 def _calculate_ratio(earned, possible):
```

Comment:
May I please suggest the following in order to close out all loose ends on this issue?  Remember that we would like to move all rounding logic of grades into a single place. 

1. Create a new function called `compute_percent`, which takes `earned` and `possible` and returns the rounded value.  Put that function in https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/scores.py
2. Change [score_for_chapter](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/course_grade.py#L111) to be `chapter_percentage`.  And update it to return the rounded percentage by calling `compute_percent`.
3. Update `get_entrance_exam_score_ratio` in this file to use the updated `chapter_percentage` function.
4. You can now eliminate `_calculate_ratio`.
5. Update the front-end code (in progress.html and progress_graph.js) to use the public functions in the Grades layer to get rounded percentage values - rather than duplicating that logic.

Whether these changes are made within this PR or in a subsequent PR, it's up to the team.  However, in order for the ticket to be considered complete, the code needs to be refactored as such for better future maintenance.  Thanks.

Reply:
@nasthagiri 
The changes you suggested have been made in subsequent commits to this PR.
also what should we be doing about rounding of subsection grade's effect on the overall grade?